### PR TITLE
Consolidated information about operator overloading

### LINKED ISSUE #6626: [operator overload] Please add a complete example and more informations
@TobiasSekan commented on [Sat Jul 21 2018](https://github.com/dotnet/docs.de-de/issues/91)

Hello, please add 
1. A complete example with all possible operators
2. Information about the opposite operators (e.g. `<<` need `>>` , `==` need `!=` )
3. `GetHashCode()` and `Equals(object a)` should be implement when compare operators are used
4. Links to important interfaces for operators `IComparable`, `IConvertible`, `IEquatable` `IFormattable` etc.

---
#### Dokumentdetails

⚠ *Bearbeiten Sie diesen Abschnitt nicht. Er ist für die Verknüpfung von docs.microsoft.com zum GitHub-Artikel erforderlich.*

* ID: 247c7eb1-fb6c-8135-71c8-5f5e50fcb8ed
* Version Independent ID: 7a3a204e-445e-d434-8141-4ed5eb409213
* Content: [operator (C#-Referenz)](https://docs.microsoft.com/de-de/dotnet/csharp/language-reference/keywords/operator)
* Content Source: [docs/csharp/language-reference/keywords/operator.md](https://github.com/dotnet/docs.de-de/blob/live/docs/csharp/language-reference/keywords/operator.md)
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by pkulikov**
Fixes #6626
Contributes to #12924 
Supported by dotnet/samples#1057

### CODE REVIEW

**Comment by BillWagner** [comment]
closing and reopening for a fresh build

File: docs/csharp/language-reference/operators/operator-overloading.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  - "operator overloading [C#]"
+---
+# Operator overloading (C# reference)
+
+A user-defined type can overload a predefined C# operator. That is, a type can provide the custom implementation of an operation when one or both operands are of that type. The [Overloadable operators](#overloadable-operators) section shows which C# operators can be overloaded.
+
+Use the `operator` keyword to declare an operator. An operator declaration must satisfy the following rules:
+
+- It includes both a `public` and a `static` modifier.
+- Unary operator takes one parameter. Binary operator takes two parameters. In each case, at least one parameter must have type `T` or `T?` where `T` is the type that contains the operator declaration.
```
Comment (BillWagner) [inline docs/csharp/language-reference/operators/operator-overloading.md]: nit:

```suggestion
- A unary operator takes one parameter. A binary operator takes two parameters. In each case, at least one parameter must have type `T` or `T?` where `T` is the type that contains the operator declaration.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
I found one small nit in word choice, and then this is ready to merge @pkulikov 

Thank you again for continue to help us improve our docs!

File: docs/csharp/language-reference/operators/operator-overloading.md (hunk 1 above)
Comment (pkulikov) [reply docs/csharp/language-reference/operators/operator-overloading.md]: @BillWagner thanks! I've applied your suggestion

**Review by BillWagner — APPROVED** [review]

**Pull request merged**